src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,220 @@
+"""
+Extraction algorithms to compute the statistics from a sequence of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.core import TelescopeComponent
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core.traits import (
+    Int,
+    List,
+)
+
+class StatisticsExtractor(TelescopeComponent):
+    """Base StatisticsExtractor component"""
+
+    sample_size = Int(
+        2500,
+        help="Size of the sample used for the calculation of the statistical values",
+    ).tag(config=True)
+    image_median_cut_outliers = List(
+        [-0.3, 0.3],
+        help="""Interval of accepted image values \\
+                (fraction with respect to camera median value)""",
+    ).tag(config=True)
+    image_std_cut_outliers = List(
+        [-3, 3],
+        help="""Interval (number of std) of accepted image standard deviation \\
+                around camera median value""",
+    ).tag(config=True)
+
+    def __init__(self, subarray, config=None, parent=None, **kwargs):
+        """
+        Base component to handle the extraction of the statistics
+        from a sequence of charges and pulse times (images).
+
+        Parameters
+        ----------
+        kwargs
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+
+    def __call__(
+        self, dl1_table, masked_pixels_of_sample=None, col_name="image"
+    ) -> list:
+        """
+        Call the relevant functions to extract the statistics
+        for the particular extractor.
+
+        Parameters
+        ----------
+        dl1_table : ndarray
+            dl1 table with images and timestamps stored in a numpy array of shape
+            (n_images, n_channels, n_pix).
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels that are not available for processing
+        col_name : string
+            column name in the dl1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and validity ranges
+        """
+
+        # in python 3.12 itertools.batched can be used
+        image_chunks = (
+            dl1_table[col_name].data[i : i + self.sample_size]
+            for i in range(0, len(dl1_table[col_name].data), self.sample_size)
```

Comment:
@mexanick about the overlapping sample windows. In principle, we could introduce a `self.sample_shift` here:
```
        image_chunks = (
            dl1_table[col_name].data[i : i + self.sample_size]
            for i in range(0, len(dl1_table[col_name].data), self.sample_shift)
        )
```
same for the time_chunks. What do you think?